route.go

```diff
@@ -48,8 +56,24 @@ func shorten(kv KVS, hasher Hasher) http.HandlerFunc {
 	}
 }
 
-func stats() http.HandlerFunc {
+func statistics(state *State) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		w.Write([]byte("TBD"))
+		updateStats(state, Statistics)
+
+		var b strings.Builder
+		for _, v := range state.Handlers {
+			fmt.Fprintf(&b, "  %v: %v\n", v.name, atomic.LoadUint64(&v.count))
+		}
+		hadlers := b.String()
+		urls := len(state.MyKVS.storage)
+		redirects := atomic.LoadUint64(&state.Stats.Redirects.Success)
+
+		s := fmt.Sprintf("URLs: %v\nRedirect: %v\nHandler:\n%v", urls, redirects, hadlers)
+		w.Write([]byte(s))
```

Comment:
What you did with the string builder above is also possible here. You could directly pass the strings builder to `fmt.Fprint` since it implements `io.Writer`.

That is exactly the same for `http.ResponseWriter`

Reply:
OK cool! Thus, do you think that the following could be more readable?

```
fmt.Fprintf(w, "URLs: %v\n", len(state.MyKVS.storage))
fmt.Fprintf(w, "Redirect: %v\n", atomic.LoadUint64(&state.Stats.Redirects.Success))
fmt.Fprintf(w, "Handler:\n")
for _, v := range state.Handlers {
    fmt.Fprintf(w, "  %v: %v\n", v.name, atomic.LoadUint64(&v.count))
}
```

Reply:
yes